path.go

```diff
@@ -370,3 +479,134 @@ func RemoveEscapeChar(word string) string {
 	}
 	return word
 }
+
+func getParamConstraintType(constraintPart string) TypeConstraint {
+	switch constraintPart {
+	case "int":
+		return intConstraint
+	case "bool":
+		return boolConstraint
+	case "float":
+		return floatConstraint
+	case "alpha":
+		return alphaConstraint
+	case "guid":
+		return guidConstraint
+	case "minLen":
+		return minLengthConstraint
+	case "maxLen":
+		return maxLengthConstraint
+	case "exactLen":
+		return exactLengthConstraint
+	case "betweenLen":
+		return BetweenLengthConstraint
+	case "min":
+		return minConstraint
+	case "max":
+		return maxConstraint
+	case "range":
+		return rangeConstraint
+	case "datetime":
+		return datetimeConstraint
+	case "regex":
+		return regexConstraint
+	default:
+		return noConstraint
+	}
+
+}
+
+func (c *Constraint) CheckConstraint(param string, matcher utils.RegexMatch) bool {
+	var err error
+	var num int
+
+	// check data exists
+	needOneData := []TypeConstraint{minLengthConstraint, maxLengthConstraint, exactLengthConstraint, minConstraint, maxConstraint, datetimeConstraint, regexConstraint}
+	needTwoData := []TypeConstraint{BetweenLengthConstraint, rangeConstraint}
+
+	for _, data := range needOneData {
+		if c.ID == data && len(c.Data) == 0 {
+			return false
+		}
+	}
+
+	for _, data := range needTwoData {
+		if c.ID == data && len(c.Data) < 2 {
+			return false
+		}
+	}
+
+	switch c.ID {
+	case intConstraint:
+		_, err = strconv.Atoi(param)
+	case boolConstraint:
+		_, err = strconv.ParseBool(param)
+	case floatConstraint:
+		_, err = strconv.ParseFloat(param, 32)
+	case alphaConstraint:
+		for _, r := range param {
+			if !unicode.IsLetter(r) {
+				err = errors.New("")
```

Comment:
please come up with something else for the error messages

would say that we create predefined errors with placeholders, which can be overlaid from the outside and then rendered in this function

the errors must then be passed up and output 
of course only when the route is normally matched without contrains check and then the contrains will be checked and any errors will be printed out 

otherwise you leave the errors out of the contrains check completely